src/ContaoCommunityAlliance/DcGeneral/Contao/View/Contao2BackendView/ContaoWidgetManager.php

```diff
@@ -467,118 +212,23 @@ public function getWidget($property, PropertyValueBag $inputValues = null)
             );
         }
 
-        $event = new BuildWidgetEvent($environment, $this->model, $propertyDefinitions->getProperty($property));
-
-        $dispatcher->dispatch($event::NAME, $event);
-        if ($event->getWidget()) {
-            return $event->getWidget();
-        }
-
-        $propInfo  = $propertyDefinitions->getProperty($property);
-        $propExtra = $propInfo->getExtra();
-        $varValue  = $this->decodeValue($property, $this->model->getProperty($property));
-        $strClass  = $this->getWidgetClass($propInfo);
-
-        if ((isset($propExtra['rgxp']) && in_array($propExtra['rgxp'], array('date', 'time', 'datim')))
-            && empty($propExtra['mandatory'])
-            && is_numeric($varValue) && $varValue == 0
-        ) {
-            /*
-                FIXME TEMPORARY WORKAROUND! To be fixed in the core:
-                @see \Widget::getAttributesFromDca()
-            */
-
-            $varValue = '';
-        }
-
-        // OH: why not $required = $mandatory always? source: DataContainer 226.
-        // OH: the whole prepareForWidget(..) thing is an only mess
-        // Widgets should parse the configuration by themselves, depending on what they need.
-        $propExtra['required'] = ($varValue == '') && !empty($propExtra['mandatory']);
-
-        if ($inputValues) {
-            $values = new PropertyValueBag($inputValues->getArrayCopy());
-            $model  = clone $this->model;
-            $model->setId($this->model->getId());
-            $this->environment->getController()->updateModelFromPropertyBag($model, $values);
-        } else {
-            $model = $this->model;
-        }
```

Comment:
I think that the model cloning could be done before triggering the event. So not every own subscriber has to do it again (and then there is no need to pass the `PropertyValueBag` to the `BuildWidgetEvent` at all).